### PR TITLE
feat: Helper Text for form fields

### LINKED ISSUE #612: Landscape Profile Helper Text
image

## Description
Provide information to the user about the meaning behind each question field in the landscape profile form. 

## Additional context
* [Design in Figma](https://www.figma.com/file/HsAlxOnHkABJ3IyT2D0SRW/Create%2FUpdate-Landscape-Profile_V2-R1.2?node-id=1712%3A2842&t=xdRi4Nj5q6fu4W6Q-0)
* [Helper Text ](https://docs.google.com/spreadsheets/d/1q0PbMgG24Rkvnby2Oc4lfh1IT-W8aUmDLDTMAGrGZbY/edit?usp=sharing)

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Helper Text for form fields

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added
- [x] English strings reviewed and copyedited
- [x] Strings are localized
- [ ] Verified on mobile
- [ ] Verified on desktop
- [ ] Verified accessibility ([Accessibility - Web applications](https://docs.google.com/document/d/1VwPDyLqw7r_iISMgDZr1FdAmAqqOIIVIxQqsYA-HaHE/edit?usp=sharing))

### Related Issues
Fixes #612 

### Verification steps

image
Screen Shot 2022-12-15 at 12 57 22 PM
Screen Shot 2022-12-15 at 12 57 09 PM

### CODE REVIEW

File: src/landscape/components/LandscapeForm/AffiliationStep.js [hunk 1]
```diff
@@ -45,6 +47,9 @@ const FORM_FIELDS = [
   {
     name: 'partnership.year',
     label: 'landscape.form_profile_partnership_year',
+    helperText: {
+      i18nKey: 'landscape.form_profile_partnership_year_helper_text',
+    },
```
Comment (paulschreiber) [inline src/landscape/components/LandscapeForm/AffiliationStep.js:52]: question: do you anticipate additional keys inside `helperText`?

File: src/landscape/components/LandscapeForm/KeyInfoUpdate.test.js [hunk 2]
```diff
@@ -109,6 +109,33 @@ test('KeyInfoUpdate: Fill form', async () => {
...
+test('KeyInfoUpdate: Helper Text', async () => {
+  terrasoApi.requestGraphQL.mockReturnValue(
+    Promise.resolve({
+      landscapes: {
+        edges: [
+          {
+            node: {
+              name: 'Landscape Name',
+              description: 'Landscape Description',
+              website: 'www.landscape.org',
```
Comment (paulschreiber) [inline src/landscape/components/LandscapeForm/KeyInfoUpdate.test.js:121]:
```suggestion
              website: 'https://www.landscape.org',
```

File: src/localization/locales/en-US.json [hunk 3]
```diff
@@ -585,7 +607,8 @@
         "validation_field_max_length": "$t({{path}}) must be fewer than {{max}} characters.",
         "validation_array_min": "$t({{path}}) must have at least {{min}} item.",
         "validation_number_integer": "$t({{path}}) must be an integer.",
-        "validation_number_max": "$t({{path}}) must be less than {{max, number()}}."
+        "validation_number_max": "$t({{path}}) must be less than {{max, number()}}.",
+        "helper_text_info_label": "[TODO] Help for: $t({{label}})"
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:611]: question: how is this used?

File: src/landscape/components/LandscapeForm/AffiliationStep.js (hunk 1 above)
Comment (josebui) [reply src/landscape/components/LandscapeForm/AffiliationStep.js:52]: @paulschreiber More than additional keys the helper text component has more parameters that can be used to cutomize how it works

File: src/localization/locales/en-US.json (hunk 3 above)
Comment (josebui) [reply src/localization/locales/en-US.json:611]: @paulschreiber This is the aria-label for the Info button that is just an icon, so it needs an aria-label

Comment (paulschreiber) [inline src/localization/locales/en-US.json:611]:
```suggestion
        "helper_text_info_label": "Information about $t({{label}})"
```

File: src/landscape/components/LandscapeForm/KeyInfoUpdate.test.js [hunk 4]
```diff
@@ -109,6 +109,33 @@ test('KeyInfoUpdate: Fill form', async () => {
...
+            },
+          },
+        ],
+      },
+    })
+  );
+  await setup('Ecuador');
+
+  const helpButton = screen.getByRole('button', {
+    name: '[TODO] Help for: Name',
```
Comment (paulschreiber) [inline src/landscape/components/LandscapeForm/KeyInfoUpdate.test.js:132]:
```suggestion
    name: 'Information about Name',
```

**Review by paulschreiber — APPROVED** [review]

File: src/landscape/components/LandscapeForm/ProfileStep.js [hunk 5]
```diff
@@ -84,7 +84,7 @@ const AreaTypesHelperText = () => {
 
 const EcosystemTypesHelperText = () => {
   return (
-    <Box sx={{ p: 2 }}>
+    <Box sx={{ p: 2, maxWidth: '40rem' }}>
```
Comment (paulschreiber) [inline src/landscape/components/LandscapeForm/ProfileStep.js:87]: suggestion: let's apply this maxWidth to all helper text boxes.

**Pull request merged**